src/vaadin-text-field-mixin.html

```diff
@@ -445,6 +457,33 @@
       }
     }
 
+    _hintChanged(hint) {
+      if (hint !== '' && hint !== null) {
```

Comment:
Looks like text-field-mixin has `_setOrToggleAttribute` that you could probably use in these three functions instead of having to repeat the logic